sdk/src/main/java/org/coralibre/android/sdk/fakegms/tasks/TaskAutorunOnceListenersThere.java

```diff
@@ -1,42 +1,78 @@
 package org.coralibre.android.sdk.fakegms.tasks;
 
+import androidx.annotation.NonNull;
+
+/**
+ * A class that waits until a success and a failure listener have been set and then runs the
+ * {@link #runInternal()} method.
+ *
+ * @param <T> the result type of the task
+ * @deprecated This subclass makes some very fragile assumptions: 1. It will run the
+ * underlying task on the same thread that sets the last listener, quite possibly the UI thread.
+ * 2. It requires both listeners to be set. 3. It assumes there won't be more than one listener per
+ * type. 4. Listeners added after the task has been run won't get called. 5. The runInternal()
+ * method is expected to call the listeners.
+ */
+@Deprecated
```

Comment:
I think this class can be deleted. Perhaps just create an additional commit which only deletes the class (after the changes introduced by you), so in case that we still need it it can simply be reverted.

Reply:
I think so too, I just wasn't sure whether I caught all usages yet.